### ISSUE TITLE
Unable to create Role and RoleBinding using kubernetes.client.handlers

### ISSUE THREAD

**Issue by hmodi23** [fabric8io/kubernetes-client#1185]
Hello,

I see that there are already handlers available for creating Role and RoleBinding as follows:

KubernetesRoleHandler - 
https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/KubernetesRoleHandler.java
KubernetesRoleBindingHandler- 
https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/KubernetesRoleBindingHandler.java

However, the latest version of kubernetes-client doesn't have these handlers:
https://mvnrepository.com/artifact/io.fabric8/kubernetes-client/4.0.3

Can we get these handlers out soon? When's the next release planned for kubernetes-client which will have these handlers available in it? In the meantime how do I access the handlers?

Thanks!

**Comment by rohanKanojia** [comment]
@hmodi23 : Hi, we are going to release kubernetes-client within a few days, I would look into this just before making the release

**Comment by hmodi23** [comment]
Hello @rohanKanojia thanks! That would be great since, I am working on a POC which will require this.

**Comment by rohanKanojia** [comment]
@hmodi23 : bdw, how are you trying to create Role and RoleBindings? Could you please share code snippet which is not working for you? We have integration tests for [KubernetesRoleBinding](https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-itests/src/test/java/io/fabric8/kubernetes/KubernetesRoleBindingIT.java) and [KubernetesRole](https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-itests/src/test/java/io/fabric8/kubernetes/KubernetesRoleIT.java). Are you doing it like this?

**Comment by hmodi23** [comment]
Hello @rohanKanojia , sure so I have a unit test case which creates KubernetesRole and KubernetesRoleBinding as follows:

@Test
    public void test() throws InterruptedException {

        /**
         *
         * kind: Role
         * apiVersion: rbac.authorization.k8s.io/v1
         * metadata:
         *   namespace: default
         *   name: pod-reader
         * rules:
         * - apiGroups: [""] # "" indicates the core API group
         *   resources: ["pods"]
         *   verbs: ["get", "watch", "list"]
         *
         */

        KubernetesPolicyRule kubernetesPolicyRule = new KubernetesPolicyRuleBuilder()
                .withApiGroups("")
                .withResources("pods")
                .withVerbs("get", "watch", "list")
                .build();

        KubernetesRole kubernetesRole = new KubernetesRoleBuilder()
                .withNewMetadata()
                .withNamespace("default")
                .withName("pod-reader")
                .endMetadata()
                .withRules(kubernetesPolicyRule)
                .build();

        /**
         * # This role binding allows "jane" to read pods in the "default" namespace.
         * kind: RoleBinding
         * apiVersion: rbac.authorization.k8s.io/v1
         * metadata:
         *   name: read-pods
         *   namespace: default
         * subjects:
         * - kind: User
         *   name: jane # Name is case sensitive
         *   apiGroup: rbac.authorization.k8s.io
         * roleRef:
         *   kind: Role #this must be Role or ClusterRole
         *   name: pod-reader # this must match the name of the Role or ClusterRole you wish to bind to
         *   apiGroup: rbac.authorization.k8s.io
         */

        KubernetesRoleBinding kubernetesRoleBinding = new KubernetesRoleBindingBuilder()
                .withNewMetadata()
                .withName("read-pods")
                .withNamespace("default")
                .endMetadata()
                .addToSubjects(0, new KubernetesSubjectBuilder()
                        .withApiGroup("rbac.authorization.k8s.io")
                        .withKind("User")
                        .withName("jane")
                        .build()
                )
                .withRoleRef(new KubernetesRoleRefBuilder()
                        .withApiGroup("rbac.authorization.k8s.io")
                        .withKind("Role")
                        .withName("pod-reader")
                        .build()
                )
                .build();

        try {
            kubernetesClient.resource(kubernetesRole).createOrReplace();
            assertResource(kubernetesRole);

            kubernetesClient.resource(kubernetesRoleBinding).createOrReplace();
            assertResource(kubernetesRoleBinding);

        } finally {
            deleteRoleBinding(kubernetesRoleBinding);
            deleteRole(kubernetesRole);
        }
    }

So, in order to push the KubernetesRole and KubernetesRoleBinding I will require the corresponding KubernetesRoleHandler and KubernetesRoleBindingHandler.

**Comment by hmodi23** [comment]
Had a couple of more questions:

1. Are there going to be similar handlers for KubernetesClusterRole and KubernetesClusterRoleBinding in the next release? 
2. Is there a way to list all the users/roles/rolebindings? I know we have "kubectl get clusterrole" and "kubectl get clusterrolebinding" for getting KubernetesClusterRole and KubernetesClusterRoleBinding.

Thanks!

**Comment by rohanKanojia** [comment]
@hmodi23 : Since #1151 is in master, yes. For the second question, Have you tried these?:
#### For listing RoleBindings
```
client.rbac().kubernetesRoleBindings().inNamespace("xyz").list()
```
#### For listing Roles
```
client.rbac().kubernetesRoles().inNamespace("xyz").list()
```
#### For listing users (in case of Openshift)
```
client.users().list();
```

**Comment by rohanKanojia** [comment]
@hmodi23 : Could you please try your above test case with SNAPSHOT version of client? In my local SNAPSHOT jar these handlers are coming, there might be some problem while making last release ..

**Comment by rohanKanojia** [comment]
@hmodi23 : Hi, we have just released kubernetes-client [v4.0.4](http://repo2.maven.org/maven2/io/fabric8/kubernetes-client/4.0.4/ 
). I checked and all handlers you were talking about were available. Could you please check again?

**Comment by hmodi23** [comment]
Thanks @rohanKanojia for the release! I just imported the new v4.0.4 and it worked just fine! Also, thanks for providing the ways to look up Roles and RoleBindings, appreciate it!
